pypika/terms.py

```diff
@@ -611,7 +611,10 @@ class Array(Tuple):
     def get_sql(self, **kwargs: Any) -> str:
         dialect = kwargs.get("dialect", None)
         values = ",".join(term.get_sql(**kwargs) for term in self.values)
-        sql = ("ARRAY[{}]" if dialect in (Dialects.POSTGRESQL, Dialects.REDSHIFT) else "[{}]").format(values)
+        if dialect in (Dialects.POSTGRESQL, Dialects.REDSHIFT):
+            sql = "ARRAY[{}]".format(values) if len(values) > 0 else "'{}'"
+        else:
+            sql = "[{}]".format(values)
```

Comment:
What do you think of declaring the default SQL as `sql = "[{}]".format(values)` and only if we have PostgreSQL or Redshift we change it.

like this:
```python
        sql = "[{}]".format(values)
        if dialect in (Dialects.POSTGRESQL, Dialects.REDSHIFT):
            sql = "ARRAY[{}]".format(values) if len(values) > 0 else "'{}'"
        return format_alias_sql(sql, self.alias, **kwargs)
```

Reply:
Is this a very specific thing where `{}` is considered an empty untyped array for postgres? As mentioned before, a test (or multiple?) could provide some clarity (and an opportunity to describe the behaviour).

Reply:
Based off of https://www.postgresql.org/docs/current/arrays.html, it seems that `'{}'` is a literal array constant just like `'{1, 2, 3}'` versus using the `ARRAY` constructor syntax `ARRAY[20000, 25000, 25000, 25000]`. Using the literal syntax seems to be preferred, as the ARRAY constructor is only mentioned offhand at the end of the section.

I tried switching the Array.get_sql to use the `'{val1, val2, val3}'` syntax but for nested arrays it generates extra single quotes: `'{'{1,2,3}','{4,5,6}'}'`. I'm guessing this is why the ARRAY constructor was used in the first place. It might make most sense to do what I originally had, with @airtonzanon 's change as it properly handles nested arrays, even though the PSQL documentation doesn't have `ARRAY` used in each inner array. I added two new array tests to check these cases. I ran the queries on a PSQL instance to check.